assets/src/editor/events/hooks/use-event-venue-entity.js

```diff
@@ -0,0 +1,55 @@
+import { useSelect } from '@wordpress/data';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+const DEFAULT_OBJECT = {
+	venueEntity: null,
+	venueEntityLoaded: false,
+};
+
+/**
+ * @typedef { Object }         VenueResponse
+ * @property {BaseEntity|null} venueEntity       The instance of the venue or
+ *                                               null.
+ * @property {boolean}         venueEntityLoaded Whether the venue has been
+ *                                               loaded or not.
+ */
+
+/**
+ * A custom hook for retrieving the venue related to the given event
+ *
+ * @param {BaseEntity} eventEntity  An instance of an event entity.
+ *
+ * @return {VenueResponse} An object containing details about the retrieved
+ *                         related venue to the event.
+ */
+const useEventVenueEntity = ( eventEntity ) => {
```

Comment:
I think one of the things I'd like to do in a separate pull(s) when the new editor gets close to "done" (there's a number of refactors to do yet) is review what can go into other modules.  We may want to create a new package module named `@eventespresso/component-hooks` (as opposed to just "hooks" to avoid confusion with WordPress js hooks) for holding hooks like this which have a broader application.

Reply:
totally agree